cloud/amazon/ec2_vpc_peer.py

```diff
@@ -0,0 +1,312 @@
+    peer_id = module.params.get('peering_id')
+    vpc_peering_connection = resource.VpcPeeringConnection(peer_id)
+    return vpc_peering_connection.status['Message']
+
+
+def create_peer_connection(client, module):
+    changed = False
+    vpc_id = module.params.get('vpc_id')
+    peer_vpc_id = module.params.get('peer_vpc_id')
+    peer_owner_id = module.params.get('peer_owner_id', False)
+    peering_conns = describe_peering_connections(vpc_id, peer_vpc_id, client)
+    for peering_conn in peering_conns['VpcPeeringConnections']:
+        if is_active(peering_conn):
+            return (False, peering_conn['VpcPeeringConnectionId'])
+        if is_pending(peering_conn):
+            return (False, peering_conn['VpcPeeringConnectionId'])
+    if not peer_owner_id:
+        try:
+            peering_conn = client.create_vpc_peering_connection(VpcId=vpc_id, PeerVpcId=peer_vpc_id)
+            return (True, peering_conn['VpcPeeringConnection']['VpcPeeringConnectionId'])
+        except botocore.exceptions.ClientError as e:
+            module.fail_json(msg=str(e))            
+    else:
+        try:
+            peering_conn = client.create_vpc_peering_connection(VpcId=vpc_id, PeerVpcId=peer_vpc_id, PeerOwnerId=str(peer_owner_id))
+            return (True, peering_conn['VpcPeeringConnection']['VpcPeeringConnectionId'])
+        except botocore.exceptions.ClientError as e:
+            module.fail_json(msg=str(e))            
+
+
+def accept_reject_delete(state, client, resource, module):
+    changed = False
+    peer_id = module.params.get('peering_id')
+    if state == "accept":
+        if peer_status(resource, module) == "Active":
+            return (False, peer_id)
+        try:
+            client.accept_vpc_peering_connection(VpcPeeringConnectionId=peer_id)
+            return (True, peer_id)
+        except botocore.exceptions.ClientError as e:
+            module.fail_json(msg=str(e))            
+    if state == "reject":
+        if peer_status(resource, module) != "Active":
+            try:
+                client.reject_vpc_peering_connection(VpcPeeringConnectionId=peer_id)
+                return (True, peer_id)
+            except botocore.exceptions.ClientError as e:
+                module.fail_json(msg=str(e))
+        else:
+            return (False, peer_id)                   
+    if state == "absent":
+        try:
+            client.delete_vpc_peering_connection(VpcPeeringConnectionId=peer_id)
+            return (True, peer_id)
+        except botocore.exceptions.ClientError as e:
+            module.fail_json(msg=str(e))            
+    return (changed, "")
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        region=dict(),
+        vpc_id=dict(),
+        peer_vpc_id=dict(),
+        peer_owner_id=dict(),
+        peering_id=dict(),
+        profile=dict(),
+        state=dict(default='present', choices=['present', 'absent', 'accept', 'reject'])
```

Comment:
@haad This module provides a 2 step process based on the action. If the action required is "present",  the input params needed for this would be quite different when requesting the "accept" action.

Example:  peering_id is only required for the accept, absent and reject actions and must be supplied, but this would break when requesting the present action if I added required=True.